cmse/cmse.md

```diff
@@ -0,0 +1,1725 @@
+inadvertent secure gateway.
+
+> 5	Memory in an NSC region must not contain an inadvertent SG instruction.
+
+The secure gateway veneers introduced in [7.4.3](#secure-gateway-veneers) limit
+the instructions that need to be placed in NSC regions. If the NSC regions
+contain only these veneers, an inadvertent secure gateway cannot occur.
+
+##	Development tools
+
+###	Source level security state changes
+
+Development tools are expected to provide C and assembly language support for
+interacting between the security states. Code written in C++ must use extern “C”
+linkage for any inter-state interaction. 
+
+> 6	Security state changes must be expressed through function calls and returns.
+
+This provides an interface that fits naturally with the C language.
+A function in secure code that can be called from the non-secure state through
+its secure gateway is called an _entry function_. A function call from secure
+state to the non-secure state is called a _non-secure function call_. This is
+shown in [Figure 2](#figure2).
+
+###	Executable files
+
+There are two different types of executable files, one for each security state.
+The secure state executes secure code from a _secure executable file_. The
+non-secure state executes non-secure code from a _non-secure executable file_.
+The secure and non-secure executable files are developed independently of each
+other. 
+
+> 7	A non-secure executable is unaware of security states.
+
+From the point of view of the non-secure state, a call to a secure gateway is a
+regular function call, as is the return from a non-secure function call. It is
+therefore required that a non-secure executable file can be developed using a
+toolchain that is not aware of CMSE.
+
+Developing a secure executable file requires toolchain support whenever a
+function is called from, calls, or returns to non-secure state and whenever
+memory is accessed through an address provided by the non-secure state. The
+secure code ABI is otherwise identical to the non-secure code ABI.
+
+For Armv8-M Mainline there are occasions where there is no possible
+code-generation that abides the VFP ABI rule that mandates that
+exception-control bits of the FPSCR may only be modified by specific support
+functions.  After returning from an entry call or when performing a
+nonsecure-call using the hard-float ABI the FPSCR will have been initialized
+with the secure world’s default FPSCR, the FPDSCR_S. For the soft-float ABI
+this is not the case as the vlstm and vldm instructions may be used. Armv8.1-M
+Mainline introduces instructions that enable the saving and restoring of the FP
+context. These are the VMSR, VMRS, VSTR and VLDR to system registers FPCXTNS and
+FPCXTS.
+
+The interaction between developers of secure code, non-secure code, and
+(optional) security agnostic library code is shown in [Figure 3](#figure3).
+
+![](figures/files-shared.png)
+
+<span id="figure3" class="citation-label">**Figure 3: Files shared between parties**</span>
```

Comment:
Same here, ca we place the `span` inside the caption of the image?

Reply:
It displaces the image if we do that.